pkg/controller/stacks/stack/stack.go

```diff
@@ -734,7 +723,9 @@ func (h *stackHandler) prepareDeployment(d *apps.Deployment) {
 	controllerDeployment.Spec.DeepCopyInto(&d.Spec)
 
 	// force the deployment to use stack opinionated names and service account
-	name := h.ext.Name + "-controller"
+	suffix := "-controller"
+	name, _ := truncate.Truncate(h.ext.Name, truncate.LabelValueLength-len(suffix), truncate.DefaultSuffixLength)
```

Comment:
Nitpick: I would normally have expected `truncate.LabelValueLength-len(suffix)` to be more like `truncate.LabelValueLength - len(suffix)` so the operator is easier to see, but maybe this is a go idiom I'm not familiar with yet.

Reply:
VS code is collapsing the spaces here. :man_shrugging: 

I'll split this into two lines. VS code is not changing the spacing when I do that.